### PR TITLE
In game GUI (Health, Ammo, Score)

### PR DESCRIPTION
**Pull request by TimTheToad**
## Purpose
  - First iteration of HUD
  - When a player fires any ammo type -> update corresponding  UI-element
  - When a playerScores -> update corresponding  UI-element
  - When a takes damage -> update corresponding  UI-element

## Changes
 - Added a new event type for when reload is finished.
 - Ammunition is not updated in fixed tick any longer, it now updates only when a WeaponFireEvent appears.
 - All UI-elements now dynamically updates.
 - Added a map in Weapon component that takes ammo type as key and a pair of Max ammo and current ammo as Value:
(as this pair<CurrentAmmo, AmmoCap>)
image

## Screen Captures (Optional)
Ammunition Capsules:
image

Score Board: 
image

Health Capsule:
image

Full HUD:
image

## Testing (Optional)
I've only tested this implementation in Sandbox and on multiplayer using 2 clients.

### CODE REVIEW

File: CrazyCanvas/Source/ECS/Systems/Player/WeaponSystem.cpp [hunk 1]
```diff
@@ -470,7 +481,7 @@ void WeaponSystem::TryFire(
 		}
 
 		// Fire the gun
-		weaponComponent.CurrentAmmunition--;
+		weaponComponent.WeaponTypeAmmo[ammoType].first--;
```
Comment (skumbananaaa) [inline CrazyCanvas/Source/ECS/Systems/Player/WeaponSystem.cpp:499]: Det är ganska kostsamt att hitta en entry i en map, du letar efter [ammoType] ovan, på rad 473. Bättre att spara resultat av den "sökningen" (som referens!) och sedan återanvända den igen.

File: CrazyCanvas/Source/ECS/Systems/Player/WeaponSystem.cpp [hunk 2]
```diff
@@ -327,7 +333,12 @@ void WeaponSystem::FixedTick(LambdaEngine::Timestamp deltaTime)
 				if (weaponComponent.ReloadClock < 0.0f)
 				{
 					weaponComponent.ReloadClock			= 0.0f;
-					weaponComponent.CurrentAmmunition	= AMMO_CAPACITY;
+					weaponComponent.WeaponTypeAmmo[EAmmoType::AMMO_TYPE_WATER].first = AMMO_CAPACITY;
+					weaponComponent.WeaponTypeAmmo[EAmmoType::AMMO_TYPE_PAINT].first = AMMO_CAPACITY;
```
Comment (skumbananaaa) [inline CrazyCanvas/Source/ECS/Systems/Player/WeaponSystem.cpp]: Se kommentaren ovan.

File: CrazyCanvas/Source/ECS/Systems/Player/WeaponSystem.cpp [hunk 3]
```diff
@@ -206,7 +206,12 @@ void WeaponSystem::FixedTick(LambdaEngine::Timestamp deltaTime)
 				if (weaponComp.ReloadClock < 0.0f)
 				{
 					weaponComp.ReloadClock = 0.0f;
-					weaponComp.CurrentAmmunition = AMMO_CAPACITY;
+					weaponComp.WeaponTypeAmmo[EAmmoType::AMMO_TYPE_PAINT].first = AMMO_CAPACITY;
+					weaponComp.WeaponTypeAmmo[EAmmoType::AMMO_TYPE_WATER].first = AMMO_CAPACITY;
```
Comment (skumbananaaa) [inline CrazyCanvas/Source/ECS/Systems/Player/WeaponSystem.cpp]: Se kommentaren ovan.

File: CrazyCanvas/Source/ECS/Systems/Player/WeaponSystem.cpp (hunk 1 above)
Comment (TimTheToad) [reply CrazyCanvas/Source/ECS/Systems/Player/WeaponSystem.cpp:499]: Alright, thanks you for pointing that out! =)

**Review by skumbananaaa — APPROVED** [review]

**Pull request merged**